zrp/prepare/geo_lookup.py

```diff
@@ -304,26 +304,13 @@ def transform(self, st_cty_code, save_table=True):
         data_path = join(curpath, '../data/processed/state_mapping.json')
         state_mapping = load_json(data_path)
 
-        aef = gps.transform(aef, state_mapping)
-        
-        # Fill the gaps in HN
-        aef = aef.astype({'FROMHN_RIGHT':'float', 'TOHN_RIGHT':'float'})
-        aef = aef.reset_index(drop = True)
-        aef['shifted_FROMHN_RIGHT'] = aef.groupby(['PARITY', 'ZEST_ZIP', 'ZEST_FULLNAME', 'FROMHN_LEFT'])['FROMHN_RIGHT']\
-                                         .apply(lambda x: x.shift(-1))
-        new_aef = aef[(aef['shifted_FROMHN_RIGHT'] != aef['TOHN_RIGHT'] + 2) & (aef['shifted_FROMHN_RIGHT'].notna())]
-        new_aef['FROMHN_RIGHT'] = (new_aef['TOHN_RIGHT'] + 2)
-        new_aef['TOHN_RIGHT']  = (new_aef['shifted_FROMHN_RIGHT'] -2)
-        new_aef['BLKGRPCE'] = None
-        new_aef['marker'] = 1
-        aef = pd.concat([aef,new_aef])       
+        aef = gps.transform(aef, state_mapping)    
         aef = aef.sort_values(['PARITY', 'ZEST_ZIP', 'ZEST_FULLNAME', 'FROMHN_LEFT', 'FROMHN_RIGHT'])                    
         aef = aef.reset_index(drop = True)
         
         final_cols = ['STATEFP', 'COUNTYFP', 'TRACTCE', 'BLKGRPCE', 'ZEST_FULLNAME', 'FROMHN' , 'TOHN', 'ZEST_ZIP', 
-                      'ZCTA5CE', 'ZCTA5CE10', 'FROMHN_LEFT', 'FROMHN_RIGHT', 'TOHN_LEFT', 'TOHN_RIGHT', 'PARITY', 'marker']
+                      'ZCTA5CE', 'ZCTA5CE10', 'FROMHN_LEFT', 'FROMHN_RIGHT', 'TOHN_LEFT', 'TOHN_RIGHT', 'PARITY']
```

Comment:
Could reducing this `final_cols` list help with preprocessing?

Reply:
It seems to me that this is the smallest possible column list required for the system to work – each of these columns are used in the geocoder.